src/domain/user.js

```diff
@@ -127,7 +127,35 @@ export default class User {
   }
 
   static async findManyByFirstName(firstName) {
-    return User._findMany('firstName', firstName)
+    return User._findMany('name', firstName)
+  }
+
+  static async findManyByName(searchTerm) {
+    return dbClient.user.findMany({
+      where: {
+        OR: [
```

Comment:
you're already using OR -> just change searchTerm to be an array of searchTerms and then your OR: part can be an array of DIFFERENT conditions -> you do a for loop:

```js
const OR = []
for searchTerm in searchTerms:
   const checkFirstName = {profile: {firstName: searchTerm, ...}
   const checkLastName = {profile: {lastName: searchTerm, ...}
   add checkFirstName and checkLastName to OR

now run query: findMany({where: { OR: OR}...)